spark/src/test/scala/org/apache/spark/sql/delta/DeltaProtocolVersionSuite.scala

```diff
@@ -588,6 +588,167 @@ trait DeltaProtocolVersionSuiteBase extends QueryTest
       "and writer versions 0, 1, 2, 3, 4, 5, 7. Please upgrade to a newer release."
   }
 
+  test("DeltaUnsupportedTableFeatureException - error message - table path") {
+    withTempDir { path =>
+      spark.range(1).write.format("delta").save(path.getCanonicalPath)
+      val (deltaLog, snapshot) = DeltaLog.forTableWithSnapshot(spark, path.getCanonicalPath)
+
+      var version = snapshot.version + 1
+      val protocolReaderFeatures = Protocol(
+        TABLE_FEATURES_MIN_READER_VERSION,
+        TABLE_FEATURES_MIN_WRITER_VERSION)
+        .withReaderFeatures(Seq("NonExistingReaderFeature1", "NonExistingReaderFeature2"))
+      untrackedChangeProtocolVersion(deltaLog, version, protocolReaderFeatures)
+
+      val exceptionRead = intercept[DeltaUnsupportedTableFeatureException] {
+        spark.read.format("delta").load(path.getCanonicalPath)
+      }
+      assert(exceptionRead.getMessage ==
```

Comment:
This should really be asserting on `exceptionRead.getErrorClass` and the parameter values, instead of validating the exact error message. (This is a recent development in Spark errors, but I think we should be following that pattern in new code.)

Reply:
I made the change. Could you double check?

Reply:
I did the same for InvalidProtocolVersionException